### PR TITLE
if property ID starts with a 40-digit hexadecimal guid, use the prope…

### PR DESCRIPTION
**Pull request by ajmers**
…rty name instead

### CODE REVIEW

**Comment by HWdata** [comment]
Hey, do we know when this will be merged? Thanks

File: tap_pipedrive/streams/recents/dynamic_typing/__init__.py [hunk 1]
```diff
@@ -11,6 +11,26 @@ class DynamicTypingRecentsStream(RecentsStream):
...
+    UNDERSCORE_CONSOLIDATION = re.compile(r"_+")
+
+    def sanitize_field_name(self, name):
+        result = name.lower()
+        result = self.SEPARATORS_TRANSLATION.sub('_', result) # Replace separator characters with underscores
+        result = self.COLUMN_NAME_TRANSLATION.sub('', result) # Remove all other non-alphanumeric characters
+        return self.UNDERSCORE_CONSOLIDATION.sub('_', result) # Consolidate consecutive underscores
+
+    def get_property_name(self, property):
+        if re.match(self.GUID_REGEX, property.get('key')):
```
Comment (dmosorast) [inline tap_pipedrive/streams/recents/dynamic_typing/__init__.py]: This can be `self.GUID_REGEX.match(property.get('key')`

**Comment by ajmers** [comment]
In testing these changes, I've realized there is more work to do here because we need to keep a mapping of old GUID names to new column names. That way as we receive data from the API we can properly translate it to the new column names. Currently we just throw it out.

**Comment by TomaszE** [comment]
HI Guys, do we have any update on that? atm I basically can't use the integration at all. Thank you

**Comment by timvisher** [comment]
See https://github.com/singer-io/tap-pipedrive/issues/23

The approach started in this PR would probably need to be reworked to take into account that the custom name on custom fields is not documented as being guaranteed to be unique.

**Pull request closed without merging**